test/main-spec.js

```diff
@@ -12,185 +12,451 @@
 
```

Comment:
I definitely agree about not duplicating the test code. My heart cried a little every time I did a copy & paste 😝 . I'll think about it. Perhaps we can add some scripts like these: ###### `tested-q.js`
```js
delete global.TestedPromise
global.TestedPromise = require('q').Promise
```
###### `tested-global.js`
```js
delete global.TestedPromise
global.TestedPromise = global.Promise
```
Which we can then require with [mocha's require flag](https://mochajs.org/#r---require-module-name).


Another option would be to put the specs in different spec-files (i.e. `./tests/q-spec.js`, `./tests/bluebird-spec.js`, `./tests/global-spec.js`, etc) and then expose the actual tests via a `spec-helper.js` like this:
```js
module.exports = function (Handlebars, describe, it, promiseName) {
  describe('promised-handlebars: ', function () {
    describe('using ' + promiseName, function () {
      it('should return a promise for the output with helpers resolved', function (done) {
        var template = Handlebars.compile(fixture('simple-helper.hbs'))
        return expect(template({a: 'abc', b: 'xyz'}))
          .to.eventually.equal('123 h(abc) 456 h(xyz)')
          .notify(done)
      })

      it('should work with block helpers that call `fn` while resolving a promise', function (done) {
        var template = Handlebars.compile(fixture('block-helper.hbs'))
        return expect(template({a: 'abc', b: 'xyz'}))
          .to.eventually.equal('123 b(abc) 456\n123 bi(cba) 456')
          .notify(done)
      })

      // other tests
    })
  })
}

function fixture (file) {
  var fs = require('fs')
  return fs.readFileSync(require.resolve('./fixtures/' + file), {encoding: 'utf-8'}).trim()
}
```

We could then write each spec file like this (here e.g. is `q-spec.js`):
```js
/* global describe */
/* global it */
// /* global xdescribe */
// /* global xit */

'use strict'

// Chai Setup
var chai = require('chai')
var chaiAsPromised = require('chai-as-promised')
chai.use(chaiAsPromised)
var expect = chai.expect

// Handlebars-Setup
var promisedHandlebars = require('../')
var QHandlebars = promisedHandlebars(require('handlebars'), {Promise: require('q').Promise})
require('./handlebars-test-setup')(Handlebars) // this does the registration of the helpers & partials

// Run the spec for Q
require('./spec-helper.js')(QHandlebars, describe, it, 'Q')

```

If we run it that way then we can stick with this in the `package.json`
```json
  "scripts": {
    "pretest": "standard --version || npm -g install standard",
    "test": "mocha",
   ...
```